src/arrayMethodJoin.js

```diff
@@ -4,8 +4,27 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let str = '';
+
+    for (let i = 0; i < this.length; i++) {
+      const item = this[i];
+
+      if (item !== null && item !== undefined) {
+        str += '' + separator + item;
```

Comment:
why do we need an empty string here if `str `is already a string

Reply:
then what about the first element? why do you add the separator before it?

Reply:
firstly we add separator before element, and then we are delete it on first position